packages/upload-client/src/blob.js

```diff
@@ -173,7 +173,7 @@ export async function add(
   const bytes =
     car instanceof Uint8Array ? car : new Uint8Array(await car.arrayBuffer())
   const link = await CAR.codec.link(bytes)
-  const digest = link.bytes
+  const digest = link.multihash.bytes
```

Comment:
@joaosa this was the culprit that caused test failures, link is a CID which is CID prefix + multihash which is why errors where saying digest length was greater than expected.